frontend/server/src/DAO/Identities.php

```diff
@@ -56,35 +56,62 @@ public static function findByUsername(string $username): ?\OmegaUp\DAO\VO\Identi
      */
-    public static function findByUsernameOrName(string $usernameOrName) {
-        $sql = "
-            SELECT
-                i.*
-            FROM
-                Identities i
-            WHERE
-                i.username = ? OR i.name = ?
-            UNION DISTINCT
-            SELECT DISTINCT
-                i.*
-            FROM
-                Identities i
-            WHERE
-                (
-                    i.username LIKE CONCAT('%', ?, '%') OR
-                    i.name LIKE CONCAT('%', ?, '%')
-                ) AND
-                i.username NOT REGEXP 'teams:[a-zA-Z0-9_.-]+:[a-zA-Z0-9_.-]+'
-            LIMIT 100";
-        $args = [$usernameOrName, $usernameOrName, $usernameOrName, $usernameOrName];
+    public static function findByUsernameOrName(
+        string $usernameOrName,
+        int $rowcount = 100
+    ) {
+        $sql = "SELECT
+                    sq.name,
+                    sq.username,
+                    SUM(sq.relevance) AS relevance
+                FROM (
+                    SELECT
+                        i.name,
+                        i.username,
+                        IFNULL(MATCH(name, username) AGAINST (? IN BOOLEAN MODE), 0) AS relevance
+                    FROM
+                        Identities i
+                    WHERE
+                        MATCH(name, username) AGAINST (? IN BOOLEAN MODE)
+                    UNION DISTINCT
+                    SELECT DISTINCT
+                        i.name,
+                        i.username,
+                        0 AS relevance
+                    FROM
+                        Identities i
+                    WHERE
+                        (
+                            i.username LIKE CONCAT('%', ?, '%') OR
+                            i.name LIKE CONCAT('%', ?, '%')
+                        ) AND
+                        i.username NOT REGEXP 'teams:[a-zA-Z0-9_.-]+:[a-zA-Z0-9_.-]+'
+                ) AS sq
+            GROUP BY
+                username, name
+            ORDER BY
+                relevance DESC
+            LIMIT
+                ?;";
+        $args = [
+            $usernameOrName,
+            $usernameOrName,
+            $usernameOrName,
+            $usernameOrName,
+            $rowcount,
+        ];
 
-        /** @var list<array{country_id: null|string, current_identity_school_id: int|null, gender: null|string, identity_id: int, language_id: int|null, name: null|string, password: null|string, state_id: null|string, user_id: int|null, username: string}> $rs */
+        /** @var list<array{name: null|string, relevance: float|null, username: string}> $rs */
         $rs = \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $args);
         $result = [];
-        foreach ($rs as $identityData) {
-            $result[] = new \OmegaUp\DAO\VO\Identities($identityData);
+        foreach ($rs as $user) {
+            $username = strval($user['username']);
```

Comment:
```suggestion
            $username = $user['username'];
```

esto ya debería ser cadena.

Reply:
Listo.